org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/testexecution/TestExecutorIntegrationTest.xtend

```diff
@@ -166,18 +167,64 @@ class TestExecutorIntegrationTest extends AbstractPersistenceIntegrationTest {
 
 		// when
 		for (var i = 0; i < 4 && statusList.head.equals('RUNNING'); i++) {
-				val future = longPollingRequest.get
-				val response = future.get(120, TimeUnit.SECONDS)
-				assertThat(response.status).isEqualTo(200)
-				statusList.offerFirst(response.readEntity(String))
-				response.close
+			val future = longPollingRequest.get
+			val response = future.get(120, TimeUnit.SECONDS)
+			assertThat(response.status).isEqualTo(200)
+			statusList.offerFirst(response.readEntity(String))
+			response.close
 		}
 
 		// then
 		assertThat(statusList.size).isGreaterThan(3)
 		assertThat(statusList.tail).allMatch['RUNNING'.equals(it)]
 		assertThat(statusList.head).isEqualTo('SUCCESS')
+	}
 
+	@Test
+	def void testThatStatusOfAllRunningAndTerminatedTestsIsReturned() {
+		// given
+		workspaceRoot.newFolder(userId)
+		workspaceRoot.newFile('''«userId»/gradlew''') => [
+			executable = true
+			JGitTestUtil.write(it, '''
+				#!/bin/sh
+				if [ "$3" = "runningTest" ]; then
+				  time=5
+				elif [ "$3" = "successfulTest" ]; then
+				  time=0
+				elif [ "$3" = "failedTest" ]; then
+				  time=-1
+				fi
+				sleep ${time}
+			''')
+		]
+		#['failed', 'successful', 'running'].map [ name |
+			workspaceRoot.newFile('''«userId»/«name»Test.tcl''')
+			'''«name»Test.tcl'''
+		].forEach [
+			val executionResponse = createTestExecutionRequest(it).post(null)
+			assertThat(executionResponse.status).isEqualTo(Status.CREATED.statusCode)
+		]
+
+		// when
+		val actualStatuses = createRequest('''tests/status/all''').get(new GenericType<Iterable<TestStatusInfo>>() {})
```

Comment:
for integration tests, why not check/match against the received json String? the frontend does not use java types ...